linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -55,8 +51,6 @@ private final class Analyzer(config: CommonPhaseConfig,
 
   private[this] val _errors = mutable.Buffer.empty[Error]
 
-  private val workQueue = new WorkQueue(ec)
```

Comment:
This removal causes quite a bit of disruption. Usually for per-run resources like that, we use a `private var`. Is there any reason not to do that here?

Reply:
I didn't really think about this option when I wrote this :) I think its fair to consider it.

- Using a `var`: Pros: less disruption, we already have mutable state
- Using an implicit: Pros: Closer to how execution contexts are handled, ensures workQueue isn't accessible everywhere (n.b. `postLoad` must not and cannot use `workQueue`).

I don't feel super strongly TBH.

Reply:
> * Using an implicit: Pros: Closer to how execution contexts are handled, ensures workQueue isn't accessible everywhere (n.b. `postLoad` must not and cannot use `workQueue`).

We don't really ensure `workQueue` not to be accessible where it's not supposed to, because `LoadingClass` captures the `WorkQueue`, and uses it in its `requestLink` method. So unless we rearrange `LoadingClass` so that we're sure it does not capture the `WorkQueue`, I don't think that pro is very strong.

Reply:
Ah, yes. Fair enough :)